### PR TITLE
fix(pubsub): AppSync in China regions

### LINKED ISSUE #10061: Amplify GraphQL API (AppSync) not working in China regions.
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React

### Amplify APIs

GraphQL API

### Amplify Categories

api

### Environment information

<details>

```
  System:
    OS: Windows 10 10.0.19044
    CPU: (8) x64 Intel(R) Core(TM) i5-8365U CPU @ 1.60GHz    
    Memory: 813.74 MB / 7.81 GB
  Binaries:
    Node: 16.13.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.10 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.6.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 103.0.5060.114
    Edge: Spartan (44.19041.1266.0), Chromium (103.0.1264.44)
    Internet Explorer: 11.0.19041.1566
  npmPackages:
    @aws-amplify/api: ^4.0.44 => 4.0.44 
    @testing-library/jest-dom: ^5.16.4 => 5.16.4 
    @testing-library/react: ^13.3.0 => 13.3.0 
    @testing-library/user-event: ^13.5.0 => 13.5.0
    react: ^18.2.0 => 18.2.0 (18.0.0)
    react-dom: ^18.2.0 => 18.2.0
    react-scripts: 5.0.1 => 5.0.1
    web-vitals: ^2.1.4 => 2.1.4
  npmGlobalPackages:
    @vue/cli: 4.5.15
    aws-cdk: 2.18.0
    cnpm: 6.1.1
    npm: 8.6.0
    nuxtron: 0.3.1
    wscat: 4.0.1
    yarn: 1.22.10
```

</details>

### Describe the bug

Amplify GraphQL API (AppSync) not working in China regions.

Following [this doc](https://docs.aws.amazon.com/appsync/latest/devguide/aws-appsync-real-time-create-generic-api-serverless-websocket.html), got `WebSocket connection to 'wss://dvu***********4q.appsync-api.cn-north-1.amazonaws.com.cn/graphql/realtime?header=eyJob***==&payload=e30=' failed: `.

This might be caused by here: https://github.com/aws-amplify/amplify-js/blob/de0441b4fa67409ccbc630c42890e2c58ee779fb/packages/pubsub/src/Providers/AWSAppSyncRealTimeProvider/index.ts#L64

The regex `/^https:\/\/\w{26}\.appsync\-api\.\w{2}(?:(?:\-\w{2,})+)\-\d\.amazonaws.com\/graphql$/i` treat China region endpoint as a custom domain.

### Expected behavior

No connection failed.

### Reproduction steps

1. Create AppSync realtime API in AWS China region.
2. Follow [this doc](https://docs.aws.amazon.com/appsync/latest/devguide/aws-appsync-real-time-create-generic-api-serverless-websocket.html), setup the react app and run.
3. Open browser, see console, got the error.

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### aws-exports.js

_No response_

### Manual configuration

_No response_

### Additional configuration

_No response_

### Mobile Device

_No response_

### Mobile Operating System

_No response_

### Mobile Browser

_No response_

### Mobile Browser Version

_No response_

### Additional information and screenshots

_No response_

### PR DESCRIPTION
**Pull request by DiscreteTom**
Fix #10061

#### Description of changes

Update `standardDomainPattern` to accept endpoints in China regions.

#### Issue #, if available

Fix #10061

#### Description of how you validated changes

Tested using the updated lib and China region AppSync realtime demo. Passed.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] ~~Relevant documentation is changed or added (and PR referenced)~~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by iartemiev** [comment]
@DiscreteTom thank you for opening this PR. Would you mind adding unit tests for this improvement to [this file](https://github.com/aws-amplify/amplify-js/blob/main/packages/pubsub/__tests__/AWSAppSyncRealTimeProvider.test.ts)?

**Comment by abdallahshaban557** [comment]
@DiscreteTom - can you please add a test as Ivan mentioned above?

File: packages/pubsub/__tests__/AWSAppSyncRealTimeProvider.test.ts [hunk 1]
```diff
@@ -40,6 +40,14 @@ describe('AWSAppSyncRealTimeProvider', () => {
 			);
 			expect(result).toBe(false);
 		});
+		
+		test('Non-custom domain in the amazonaws.com.cn subdomain space returns `false`', () => {
```
Comment (stocaaro) [inline packages/pubsub/__tests__/AWSAppSyncRealTimeProvider.test.ts:44]: After manually validating the structure of appsync urls in cn, recommending we add a test, which I've gone ahead and committed.

**Review by stocaaro — APPROVED** [review]

**Comment by stocaaro** [comment]
@DiscreteTom Thank you so much for this change. I've done my own validation and added a test and will work with the team to get this merged.

**Review by jimblanc — APPROVED** [review]

**Pull request merged**